src/uct/ib/base/ib_md.c

```diff
@@ -854,8 +855,8 @@ ucs_status_t uct_ib_query_md_resources(uct_component_t *component,
     }
 
     for (i = 0; i < num_devices; ++i) {
-        /* Skip non-existent and non-accessible devices */
-        if (!uct_ib_device_is_accessible(device_list[i])) {
+        /* Skip not applicable devices */
+        if (!predicate(device_list[i])) {
```

Comment:
we can call predicate after uct_ib_device_is_accessible, not instead
and then rename uct_ib_query_md_resources_with_pred to uct_ib_query_md_resources, like so:
```c
ucs_status_t uct_ib_base_query_md_resources(uct_md_resource_desc_t **resources_p,
                                        unsigned *num_resources_p,
                                        uct_ib_check_device_cb_t check_device_cb)
```